v3/src/components/graph/adornments/count/count-adornment-component.tsx

```diff
@@ -35,13 +36,24 @@ export const CountAdornment = observer(function CountAdornment(props: IAdornment
   const defaultFontSize = graphModel.adornmentsStore.defaultFontSize
   let fontSize = defaultFontSize
   const prevCellWidth = useRef(plotWidth)
-  const subPlotRegionBoundaries = useMemo(
-    () => adornmentsStore?.subPlotRegionBoundaries(instanceKey, scale) ?? [],
-    [adornmentsStore, instanceKey, scale]
-  )
-  const subPlotRegionBoundariesRef = useRef(subPlotRegionBoundaries)
   const [displayCount, setDisplayCount] = useState(<div>{textContent}</div>)
 
+  const subPlotRegionBoundaries = useCallback(() => {
+      // Sub plot regions can be defined by either bin boundaries when points are grouped into bins, or by
+      // instances of the movable value adornment. It should not be possible to have both bin boundaries and
+      // movable values present at the same time.
+      if (graphModel.pointDisplayType === "bins") {
+        const { binWidth, minBinEdge, maxBinEdge, totalNumberOfBins } = graphModel.binDetails()
+        return [
+          ...Array.from({ length: totalNumberOfBins }, (_, i) => minBinEdge + i * binWidth),
```

Comment:
Clever -- probably worth a comment, however, since the syntax is not immediately obvious.